### DISCUSSION TITLE
Extend Delivery API to get user/member info

### DISCUSSION THREAD

**Discussion by bjarnef** [umbraco/Umbraco-CMS#17508] [Features and ideas]
Currently it seems there isn't any way to fetch information of current logged in member.
https://docs.umbraco.com/umbraco-cms/13.latest/reference/content-delivery-api/protected-content-in-the-delivery-api?fallback=true

It seems there's a `SetUserinfoEndpointUris()` which could be used:
https://dev.to/robinvanderknaap/setting-up-an-authorization-server-with-openiddict-part-v-openid-connect-a8j

but isn't set here:
https://github.com/umbraco/Umbraco-CMS/blob/v13/dev/src/Umbraco.Cms.Api.Common/DependencyInjection/UmbracoBuilderAuthExtensions.cs#L36-L43

I found an example here to get `userinfo`:
https://github.com/openiddict/openiddict-samples/blob/dev/samples/Velusia/Velusia.Server/Controllers/UserinfoController.cs#L28

Is it possible to configure this with your own endpoint for now?

**Comment by bjarnef** [comment]
@bergmania @kjac could you check this? 😉

**Comment by kjac** [comment]
This makes a ton of sense. As a bare minimum, we should expose the member name and email 👍 

We could also consider exposing all custom properties that aren't marked as sensitive?

**Reply by bjarnef** [reply]
At least we needed `name` and possible email/username. Node name may not always be a friendly name, e.g. it could be something like `[id] - [name]`. So it would probably make sense to expose some custom properties like `firstName`, `lastName` etc.

The other endpoints look up member identity, but doesn't return data from identity user (member).

**Comment by kjac** [comment]
PR in #17719 👍

**Comment by elit0451** [comment]
Implemented in #17719
